readme.md

```diff
@@ -133,19 +133,26 @@ const getData = async id => {
 Bluebird:
 
 ```js
-Promise.resolve([1, 2]).spread((one, two) => {
-	console.log(one, two);
-	//=> 1 2
-});
+const getLoginData = ids =>
+	Promise.resolve(ids)
+		.then(getUserLoginCount)
+		.spread((...counts) => {
+			console.log(counts); //=> 10 20
+			return [ids, counts] 
+		});
 ```
 
 Instead, use [destructuring](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment):
 
 ```js
-Promise.resolve([1, 2]).then(([one, two]) => {
-	console.log(one, two);
-	//=> 1 2
-});
+const getLoginData = ids =>
+	Promise.resolve(ids)
+		.then(getUserLoginCount)
+		.then(results => Promise.all(results))
+		.then(counts => {
+			console.log(counts); //=> 10 20
+			return [ids, counts] 
```

Comment:
Shows a single-function pattern to combine values across nested scope into the return value ` [ids, counts]`
This eliminates creating extra array/object wrappers - simply to pass ad hoc state between logical steps. 

I feel this sort of code is tricky for beginners:
```js
   .then(data => Promise.all([data, process(data)]) // not obvious if `data` is being mutated in `process()`, affecting the passed data recieved in next .then
   .then(([data, result]) => prepare(data, result));
```
If we need to return an array w/ calling args + promised result, my pattern could be followed more closely by creating the simplest generalized function:

```js
const wrapCallWithArgs = fn => (...args) => [args, fn.apply(null, args)] 
// ^^^ Helps discourage one-off arbitrary mutations - now we can reliably use this [arg, results] pattern 

// Cleaned up 
p.then(wrapCallWithArgs(process))
  .then(prepare) // prepare would need to support arrays to tighten up syntax to this
//.then((...args) => prepare(...args)) // or use this w/o change to prepare func signature



```